File: app/src/main/java/com/andraganoid/verymuchtodo/J/model/Document.java
package com.andraganoid.verymuchtodo.J.model;

import java.util.HashMap;

public class Document {

    public static final String COLLECTION_USERS = "collectionUsers";
    public static final String COLLECTION_TODO_LISTS = "collectionToDoList";
    public static final String COLLECTION_MESSAGES = "collectionMessages";
    public static final String COLLECTION_LOCATION = "collectionLocation";

    private String collection;
    private String documentName;
    private HashMap<String, Object> map;

    public Document(User user) {
        map = new HashMap<>();
        map.put("id", user.getId());
        map.put("name", user.getName());
        map.put("email", user.getEmail());
        collection = COLLECTION_USERS;
        documentName = user.getId();
    }

    public Document(TodoList todoList) {
        map = new HashMap<>();
        map.put("id", todoList.getId());
        map.put("title", todoList.getTitle());
        map.put("description", todoList.getDescription());
        map.put("emergency", todoList.isEmergency());
        map.put("completed", todoList.isCompleted());
        map.put("todoItemList", todoList.getTodoItemList());
        map.put("timestamp", todoList.getTimestamp());
        map.put("user", todoList.getUser());
        collection = COLLECTION_TODO_LISTS;
        documentName = todoList.getId();
    }

    public Document(Message message) {
        map = new HashMap<>();
        map.put("text", message.getText());
        map.put("timestamp", message.getTimestamp());
        map.put("user", message.getUser());
        map.put("id", message.getId());
        documentName = message.getId();
        collection = COLLECTION_MESSAGES;
    }

    public Document(ToDoLocation location) {
        map = new HashMap<>();
        map.put("latitude", location.getLatitude());
        map.put("longitude", location.getLongitude());
        map.put("timestamp", location.getTimestamp());
        map.put("user", location.getUser());
        collection = COLLECTION_LOCATION;
        documentName = location.getUser().getId();
    }

    public String getCollection() {
        return collection;
    }

    public String getDocumentName() {
        return documentName;
    }

    public HashMap<String, Object> getMap() {
        return map;
    }

}

File: app/src/main/java/com/andraganoid/verymuchtodo/shortVersion/model/Document.kt
package com.andraganoid.verymuchtodo.shortVersion.model

import com.andraganoid.verymuchtodo.shortVersion.util.COL_LIST

class Document(value: Any?) {

    var collection: String = ""
    var name: String = ""
    var values: Map<String?, Any?> = mapOf()

    init {
        todoList(value as TodoList)
    }

    private fun todoList(todoList: TodoList) {
        collection = COL_LIST
        name = todoList.id
        values = mapOf(
            "title" to todoList.title,
            "id" to todoList.id,
            "description" to todoList.description,
            "completed" to todoList.completed,
            "userName" to todoList.userName,
            "todoList" to todoList.todoList,
            "timestamp" to todoList.timestamp
        )
    }


}